src/QsCompiler/Transformations/CallGraph/ConcreteCallGraphWalker.cs

```diff
@@ -256,9 +258,18 @@ private void AddEdge(QsQualifiedName identifier, QsSpecializationKind kind, Type
                     var called = new ConcreteCallGraphNode(identifier, kind, typeRes);
                     var edge = new ConcreteCallGraphEdge(referenceRange);
                     this.Graph.AddDependency(this.CurrentNode, called, edge);
-                    if (!this.RequestStack.Contains(called) && !this.ResolvedNodeSet.Contains(called))
+
+                    // make sure we always build the dependencies for all specializations
```

Comment:
Is this for issue #757? Does this make PR #758 unnecessary?

Reply:
I think it should, but that needs to be confirmed.